src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -266,21 +381,7 @@ private function getDefinitionSchema(string $resourceClass, ResourceMetadata $re
             ];
         }
 
-        $attributes = $resourceMetadata->getAttributes();
-        $context = [];
-        if (isset($attributes['normalization_context']['groups'])) {
-            $context['serializer_groups'] = $attributes['normalization_context']['groups'];
-        }
-
-        if (isset($attributes['denormalization_context']['groups'])) {
-            if (isset($context['serializer_groups'])) {
-                $context['serializer_groups'] += $attributes['denormalization_context']['groups'];
-            } else {
-                $context['serializer_groups'] = $attributes['denormalization_context']['groups'];
-            }
-        }
-
-        foreach ($this->propertyNameCollectionFactory->create($resourceClass, $context) as $propertyName) {
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass, ['serializer_groups' => (null !== $groups ? $groups['groups'] : [])]) as $propertyName) {
```

Comment:
It's not ok: if `$groups` is `null`, the `'serializer_groups'` key should not exist at all. Using an empty array means "no groups", `null` means "all properties".